provd/protos/user.proto

```diff
@@ -38,5 +29,14 @@ message ValidateUsernameRequest {
 }
 
 message ValidateUsernameResponse {
-  bool valid = 1;
+  UsernameValidation username_validation = 1;
 }
+
+enum UsernameValidation {
+    OK = 0;
+    ALREADY_IN_USE = 1;
+    SYSTEM_RESERVED = 2;
+    INVALID_CHARS = 3;
+    TOO_LONG = 4;
+    EMPTY = 5;
```

Comment:
Nitpick: the indentation level is different here than in the other blocks

Reply:
Fair! I see the style guide says 2 spaces, which is what I've done everywhere else, so I'll just align this :+1: